src/physics/grid/TwodGridInterface.hh

```diff
@@ -32,10 +36,12 @@ struct TwodGridData
     //! Get the data location for a specified x-y coordinate.
     CELER_FUNCTION ItemId<real_type> at(size_type ix, size_type iy) const
     {
-        CELER_EXPECT(ix < this->x.size());
-        CELER_EXPECT(iy < this->y.size());
-        return ItemId<real_type>{this->values.front().get()
-                                 + ix * this->y.size() + iy};
+        CELER_EXPECT(ix <= this->x.size());
+        CELER_EXPECT(iy <= this->y.size());
+        size_type index = ix * this->y.size() + iy;
+
+        CELER_ENSURE(index <= this->x.size() * this->y.size());
```

Comment:
Wait why the changes to `<= size`? Shouldn't it be `<`?

Reply:
Sometimes it's useful to construct ranges by giving past-the-end values -- but I don't I ended up using it that way, so I'll revert this change too.